controllers/space/space_controller.go

```diff
@@ -121,53 +139,80 @@ func (r *Reconciler) ensureNSTemplateSet(logger logr.Logger, space *toolchainv1a
 		if errors.IsNotFound(err) {
 			logger.Info("creating NSTemplateSet on target member cluster")
 			if err := r.setStatusProvisioning(space); err != nil {
-				return err
-			}
-			if nsTmplSet, err = r.newNSTemplateSet(memberCluster.OperatorNamespace, space); err != nil {
-				return r.setStatusProvisioningFailed(logger, space, err)
+				return false, r.setStatusProvisioningFailed(logger, space, err)
```

Comment:
```suggestion
				return false, err
```
If `r.setStatusProvisioning(space)` then we should probably just return false, err instead?

Reply:
this code is no outdated, because the reason for failing was when fetching the NSTemplateTier would fail. This block has been extracted into:
```
tmplTier := &toolchainv1alpha1.NSTemplateTier{}
if err := r.Client.Get(context.TODO(), types.NamespacedName{
	Namespace: space.Namespace,
	Name:      space.Spec.TierName,
}, tmplTier); err != nil {
	return false, r.setStatusProvisioningFailed(logger, space, err)
}
```
but I believe that it's valid to set the status of the Space to `UnableToProvision` if the Tier does not exist, isn't it?

Reply:
What I was trying to say before was that if `r.setStatusProvisioning(space)` was returning an error then trying to do `r.setStatusProvisioningFailed` wouldn't make much sense because it's another status update. ie. if one status update failed, we probably shouldn't try to make another status update

But it's changed now so what you have is fine. 👍